src/icesat2_tracks/analysis_db/B03_plot_spectra_ov.py

```diff
@@ -0,0 +1,581 @@
+plt.ylabel("y_coord deviation (m)")
+
+
+F.save_light(path=plot_path, name="B03_specs_coord_check")
+
+
+def dict_weighted_mean(Gdict, weight_key):
+    """
+    returns the weighted meean of a dict of xarray, data_arrays
+    weight_key must be in the xr.DataArrays
+    """
+
+    akey = list(Gdict.keys())[0]
+    GSUM = Gdict[akey].copy()
+    GSUM.data = np.zeros(GSUM.shape)
+    N_per_stancil = GSUM.N_per_stancil * 0
+    N_photons = np.zeros(GSUM.N_per_stancil.size)
+
+    counter = 0
+    for I in Gdict.items():
+        I = I.squeeze()
+        if len(I.x) != 0:
+            GSUM += I.where(~np.isnan(I), 0) * I[weight_key]  # .sel(x=GSUM.x)
+            N_per_stancil += I[weight_key]
+        if "N_photons" in GSUM.coords:
+            N_photons += I["N_photons"]
+        counter += 1
+
+    GSUM = GSUM / N_per_stancil
+
+    if "N_photons" in GSUM.coords:
+        GSUM.coords["N_photons"] = (("x", "beam"), np.expand_dims(N_photons, 1))
+
+    GSUM["beam"] = ["weighted_mean"]
+    GSUM.name = "power_spec"
+
+    return GSUM
+
+
+G_gFT_wmean = (
+    Gk["gFT_PSD_data"].where(~np.isnan(Gk["gFT_PSD_data"]), 0) * Gk["N_per_stancil"]
+).sum("beam") / Gk["N_per_stancil"].sum("beam")
+G_gFT_wmean["N_per_stancil"] = Gk["N_per_stancil"].sum("beam")
+
+G_fft_wmean = (Gfft.where(~np.isnan(Gfft), 0) * Gfft["N_per_stancil"]).sum(
+    "beam"
+) / Gfft["N_per_stancil"].sum("beam")
+G_fft_wmean["N_per_stancil"] = Gfft["N_per_stancil"].sum("beam")
+
+
+def plot_wavenumber_spectrogram(ax, Gi, clev, title=None, plot_photon_density=True):
+    if Gi.k[0] == 0:
+        Gi = Gi.sel(k=Gi.k[1:])
+    x_lambda = 2 * np.pi / Gi.k
+    plt.pcolormesh(
+        Gi.x / 1e3, x_lambda, Gi, cmap=plt.cm.ocean_r, vmin=clev[0], vmax=clev[-1]
+    )
+
+    ax.set_yscale("log")
+
+    if plot_photon_density:
+        plt.plot(
+            Gi.x / 1e3,
+            x_lambda[-1] + (Gi.N_per_stancil / Gi.N_per_stancil.max()) * 10,
+            c="black",
+            linewidth=0.8,
+            label="NAN-density",
+        )
+        plt.fill_between(
+            Gi.x / 1e3,
+            x_lambda[-1] + (Gi.N_per_stancil / Gi.N_per_stancil.max()) * 10,
+            0,
+            color="gray",
+            alpha=0.3,
+        )
+        ax.axhline(30, color="black", linewidth=0.3)
+
+    plt.ylim(x_lambda[-1], x_lambda[0])
+    plt.title(title, loc="left")
+
+
+Gmean = G_gFT_wmean.rolling(k=5, center=True).mean()
+
+try:
+    k_max_range = (
+        Gmean.k[Gmean.isel(x=slice(0, 5)).mean("x").argmax().data].data * 0.75,
+        Gmean.k[Gmean.isel(x=slice(0, 5)).mean("x").argmax().data].data * 1,
+        Gmean.k[Gmean.isel(x=slice(0, 5)).mean("x").argmax().data].data * 1.25,
+    )
+except:
+    k_max_range = (
+        Gmean.k[Gmean.isel(x=slice(0, 20)).mean("x").argmax().data].data * 0.75,
+        Gmean.k[Gmean.isel(x=slice(0, 20)).mean("x").argmax().data].data * 1,
+        Gmean.k[Gmean.isel(x=slice(0, 20)).mean("x").argmax().data].data * 1.25,
+    )
```

Comment:
```suggestion
try:
    k_max = Gmean.k[Gmean.isel(x=slice(0, 5)).mean("x").argmax().data].data
except:
    k_max = Gmean.k[Gmean.isel(x=slice(0, 20)).mean("x").argmax().data].data

k_max_range = (k_max * 0.75, k_max * 1, k_max * 1.25)
```

`Gmean.k[Gmean.isel(x=slice(0, 5)).mean("x").argmax().data].data` could also be moved to a function but this is simpler for now.

Reply:
I applied your change